test/Engine/ProtoTest/Associative/FunctionObject.cs

```diff
@@ -497,5 +497,102 @@ def getName(f)
             thisTest.Verify("foos", new object[] {1, 2});
             thisTest.Verify("bars", new object[] {3, 4});
         }
+
+        [Test]
+        public void Test__GroupByKey1()
+        {
```

Comment:
Can you add a test that tests migrations  - There are a few immediately failing tests because there is no migration for a very old .63 node GroupBy, which was migrated to GroupByKey - check line 765 of the ``dynBaseType.cs`` migrations file, I guess this old node should be migrated to GroupByFunction as well. There are tests for this node already, but it would be good to have a migration test for GroupByKey - > GroupByFunction.